client/state/billing-data/util.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with date converted from string to a Date object.
+ */
+export const parseTransactionDate = ( transaction ) => {
+	return { ...transaction, date: new Date( transaction.date ) };
```

Comment:
Whoops, let's stick with moment. Tested some sample strings in Safari and it looks unhappy with:
```
new Date( "2016-12-12T11:22:33+0000" ).getTime() //NaN
```